### PR TITLE
(chore) change tests in 27ica/controller to expect specific error

### PR DESCRIPTION
**Pull request by bznein**
## Description

closes: #7178

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go [hunk 1]
```diff
@@ -242,11 +242,11 @@ func (suite *InterchainAccountsTestSuite) TestOnChanOpenInit() {
 					path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, chanCap, channel.Counterparty, channel.Version,
 				)
 
-				if tc.expPass {
+				if tc.expErr == nil {
 					suite.Require().Equal(TestVersion, version)
 					suite.Require().NoError(err)
 				} else {
-					suite.Require().Error(err)
+					suite.Require().ErrorContains(err, tc.expErr.Error())
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go:249]: Yeah, I think we have faced this issue other times as well... Something about how the error is wrapped, I think...

File: modules/apps/27-interchain-accounts/controller/keeper/grpc_query_test.go [hunk 2]
```diff
@@ -12,33 +12,33 @@ func (suite *KeeperTestSuite) TestQueryInterchainAccount() {
 	testCases := []struct {
 		name     string
 		malleate func()
-		expPass  bool
+		errMsg   string
```
Comment (crodriguezvega) [inline modules/apps/27-interchain-accounts/controller/keeper/grpc_query_test.go:15]: Is it not possible to use `expErr` here?

File: modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go [hunk 3]
```diff
@@ -25,46 +27,46 @@ func (suite *KeeperTestSuite) TestRegisterInterchainAccount_MsgServer() {
 
 	testCases := []struct {
 		name     string
-		expPass  bool
+		expErr   error
```
Comment (crodriguezvega) [inline modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go:30]: Nit: just for consistency with the other tests, we could just switch the order of the last two and have `expErr` last.

File: modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go [hunk 4]
```diff
@@ -25,46 +27,46 @@ func (suite *KeeperTestSuite) TestRegisterInterchainAccount_MsgServer() {
...
 		{
 			"invalid connection id",
-			false,
+			connectiontypes.ErrConnectionNotFound,
 			func() {
 				msg.ConnectionId = "connection-100"
 			},
 		},
 		{
 			"non-empty owner address is valid",
```
Comment (crodriguezvega) [inline modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go:54]: Maybe we could also move this test up together with the rest of success cases.

**Review by crodriguezvega — APPROVED** [review]
Great work, @bznein. Thank you so much for taking the initiate to refactor the tests!

File: modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go (hunk 4 above)
Comment (bznein) [reply modules/apps/27-interchain-accounts/controller/keeper/msg_server_test.go:54]: Yeah I actually wanted to do this but also wanted to keep the diff tidy! Now that it has been reviewed I think it makes more sense to move it :)

File: modules/apps/27-interchain-accounts/controller/keeper/grpc_query_test.go (hunk 2 above)
Comment (bznein) [reply modules/apps/27-interchain-accounts/controller/keeper/grpc_query_test.go:15]: I wasn't able to. We return errors with `status.Errorf` which is not the same as wrapping errors. They are not a specific error type

**Review by chatton — APPROVED** [review]
LGTM! Thanks for cleaning these up

**Pull request merged**